src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1361,6 +1362,24 @@ public static bool IsFileTypeOrUsesFileTypes(this TypeSymbol type)
             return foundType is not null;
         }
 
+        internal static string? AssociatedFileIdentifier(this NamedTypeSymbol type)
```

Comment:
>AssociatedFileIdentifier [](http://example.com/codeflow?start=32&length=24)

Why is this method among general helpers if works only on ```SourceMemberContainerTypeSymbol```?

Reply:
I think we should make it hard to use it by mistake on a wrong symbol flavor. I.e. one should get the right flavor before calling the helper. Otherwise there is a risk to get wrong answer.

Reply:
Added abstract property `NamedTypeSymbol.AssociatedSyntaxTree`. This makes this helper apply to more kinds of named types.